### ISSUE TITLE
Remove need to add RCAP specific code to a notebook

### ISSUE THREAD

**Issue by shaneporter** [att/rcloud.rcap#72]
The following code must be added (last cell?) to a notebook so that it is interpreted properly in rcap.html:

```r
library(rcloud.rcap) 
library(rcloud.web) # not sure if this is needed
rcap.result()
```

Is there any way this can be done automatically by the package?

**Comment by gaborcsardi** [comment]
Maybe the easiest is to just add a cell to the notebook from JS, containing the required function call.

**Comment by gaborcsardi** [comment]
I'll give this a shot.

**Comment by gaborcsardi** [comment]
Well, adding a cell to the notebook is not really possible....

The R code creates the dashboard currently, in the first place.

**Comment by gordonwoodhull** [comment]
I haven't tried this, but couldn't you replace your call of `rcloud.call.notebook` (`rcloud.call_notebook()` in js/rcap.js) with another function that wraps it and returns the value that you want?

`rcloud.call.notebook` just evals each cell and then returns the result of the last one. So I think that creating another function which calls `rcloud.call.notebook` but returns something else.

Alternately, I guess you could just completely change what rcap.js does with the result, or have it call some other function after call.notebook in order to get the right result (a result with some ocaps to run, basically).

Sorry I wasn't paying much attention here. Always feel free to ask me any RCloud design questions, you probably won't find it in the documentation. :wink:

**Comment by gaborcsardi** [comment]
@gordonwoodhull No worries, I made good progress.

It is not so simple, because `rcap.result()` does not return anything, but the loading of the `rcloud.rcap` package itself sets some ocaps, and then `rcap.result()` also does a bunch of things. 

But when I am running `rcap.js`, I might not even have the `rcloud.rcap` package loaded, so no rcap ocaps are defined either. Anyway, I managed to trigger the loading of the `rcloud.rcap` package, so I can just call an ocap after the notebook is evaluated. Will finish it next week.
